### PR TITLE
Add apigw-s3-terraform pattern

### PR DESCRIPTION
**Pull request by enmata**
*Issue #, if available:*
Adding apigw-s3-terraform patterrn

*Description of changes:*
Adding terraform version of apigw-s3-sam pattern
https://serverlessland.com/patterns/apigateway-rest-s3-sam

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: apigw-s3-terraform/README.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
+# REST API as an Amazon S3 proxy in API Gateway
+
+This pattern in CDK offers a boilerplate to generate an Amazon API Gateway REST API endpoint to Amazon S3 with the following methods -
```
Comment (pputhran) [inline apigw-s3-terraform/README.md:3]: This talks about CDK?

File: apigw-s3-terraform/README.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    ```
+    cd apigw-s3-terraform
+    ```
+1. From the command line, initialize terraform to  to downloads and installs the providers defined in the configuration:
+    ```
+    terraform init
+    ```
+1. From the command line, apply the configuration in the main.tf file:
+    ```
+    terraform apply
```
Comment (pputhran) [inline apigw-s3-terraform/README.md:34]: Getting the following error on terraform apply

│ Error: Invalid provider configuration
│ 
│ Provider "registry.terraform.io/hashicorp/aws" requires explicit configuration. Add a provider block to the root module and configure the provider's required arguments as described in the provider documentation.
│ 
╵
╷
│ Error: Invalid AWS Region: yes
│ 
│   with provider["registry.terraform.io/hashicorp/aws"],
│   on <input-prompt> line 1:
│   (source code not available)
│

**Review by pputhran — CHANGES_REQUESTED** [review]
This pattern is also missing example-pattern.json file. Can you check this example-pattern.json and add it?

[example](https://github.com/aws-samples/serverless-patterns/blob/main/_pattern-model/example-pattern.json)

File: apigw-s3-terraform/README.md (hunk 1 above)
Comment (enmata) [reply apigw-s3-terraform/README.md]: Thanks for the heads up, fixed on new comit

File: apigw-s3-terraform/README.md (hunk 2 above)
Comment (enmata) [reply apigw-s3-terraform/README.md:34]: Changes applied on new main.tf. Please, could you check?

**Comment by enmata** [comment]
Changes added. Please, could you check again?

**Comment by enmata** [comment]
Code has been refactored. Please, could you check again?

**Comment by ellisms** [comment]
@enmata I'm trying to test this and sending a GET to the base URL and base URL/bucket-name results in 500 errors. I can replicate the same error in the console using the 'test' button.

**Comment by ellisms** [comment]
Hi @enmata - have you reviewed the HTTP 500 error I reported?

**Comment by ellisms** [comment]
Closing this as stale.

**Pull request closed without merging**